internal/ingress/controller/status.go

```diff
@@ -93,12 +93,24 @@ func setupLeaderElection(config *leaderElectionConfig) {
 		Host:      hostname,
 	})
 
-	lock := resourcelock.ConfigMapLock{
-		ConfigMapMeta: metav1.ObjectMeta{Namespace: k8s.IngressPodDetails.Namespace, Name: config.ElectionID},
-		Client:        config.Client.CoreV1(),
-		LockConfig: resourcelock.ResourceLockConfig{
-			Identity:      k8s.IngressPodDetails.Name,
-			EventRecorder: recorder,
+	objectMeta := metav1.ObjectMeta{Namespace: k8s.IngressPodDetails.Namespace, Name: config.ElectionID}
+	resourceLockConfig := resourcelock.ResourceLockConfig{
+		Identity:      k8s.IngressPodDetails.Name,
+		EventRecorder: recorder,
+	}
+
+	// TODO: If we upgrade client-go to v0.24 then we can only use LeaseLock.
```

Comment:
When was LeaseLock created? If in v1.21, we can leave in the TODO here that part of the stabilization code will contain only LeaseLock and no ConfigMap anymore.

Reply:
Agreed. Let's move forward with this, and plan the new release after freeze without the MultiLock